connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/InboundConnectorContextImpl.java

```diff
@@ -66,6 +67,11 @@ public void correlate(Object variables) {
     correlationHandler.correlate(definition, variables);
   }
 
+  @Override
+  public CorrelationResult correlateWithResult(Object variables) {
+    return correlationHandler.correlate(definition, variables);
+  }
```

Comment:
I think we can return a `CorrelationResult` for the existing code without breaking any contracts right?

Reply:
On the source code level - yeah, but the resulting byte code on the caller side will become incompatible.

I tested it with a simpler example by compiling the java class, compiling another class where it was used, then replacing the first class without recompiling the second one - it throws a `NoSuchMethodException` even though I only changed the return type.

So this will make all connectors that are already compiled incompatible with the new runtime.

Reply:
Did you change the return type from `void` to a type or did you change the type lets say from `int` to `String`?

Reply:
From void to a type